code/game/objects/mail.dm

```diff
@@ -0,0 +1,233 @@
+/obj/item/envelope
+	name ="broken letter"
+	desc = "We just got a letter, we just got a letter, we just got a letter- wonder who its from."
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mail_misc"
+	item_state = "paper"
+	drop_sound = 'sound/items/handling/paper_drop.ogg'
+	pickup_sound =  'sound/items/handling/paper_pickup.ogg'
+	var/list/possible_contents = list()
+	var/list/job_list = list()
+	var/mob/living/recipient
+	var/has_been_scanned = FALSE
+
+/obj/item/envelope/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is licking a sharp corner of the envelope. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(loc, 'sound/effects/-adminhelp.ogg', 50, TRUE, -1)
+	return BRUTELOSS
+
+/obj/item/envelope/attack_self(mob/user)
+	if(user.mind.current != recipient.mind.current)
```

Comment:
This will also runtime if something mindless calls attack_self.
```suggestion
/obj/item/envelope/attack_self(mob/user)
	if(!user.mind)
		return
	if(user.mind.current != recipient.mind.current)
```

Reply:
Done!